src/nupic/research/connections.py

```diff
@@ -242,6 +259,48 @@ def updateSynapsePermanence(self, synapse, permanence):
     # Update indexes
     self._synapsesForPresynapticCell[newData.presynapticCell][synapse] = newData
 
+  def computeActivity(self, activeInput, activePermanenceThreshold,
+                      activeSynapseThreshold, matchingPermananceThreshold,
+                      matchingSynapseThreshold):
+    """
+    Computes active and matching segments given the current active input.
+
+    @param activeInput                 (set)   currently active cells
+    @param activePermanenceThreshold   (float) permanence threshold for a synapse to be considered active
+    @param activeSynapseThreshold      (int)   number of synapses needed for a segment to be considered active
+    @param matchingPermananceThreshold (float) permanence threshold for a synapse to be considered matching
+    @param matchingSynapseThreshold    (int)   number of synapses needed for a segment to be considered matching
+
+    @return (tuple) Contains:
+                      `activeSegments`         (list),
+                      `matchingSegments`        (list),
+
+    Notes:
+      activeSegments and matchingSegments are sorted by their corresponding presynaptic cell.
+    """
+
+    numActiveSynapsesForSegment = defaultdict(int)
+    numMatchingSynapsesForSegment = defaultdict(int)
+    activeSegments = set()
+    matchingSegments = set()
```

Comment:
This is building up a sorted list. It should be faster to just use an unsorted list and then sort it at the end.

i.e. we can just change these to lists.